x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -276,6 +276,9 @@ export function XYChart({
         legendPosition={legend.position}
         showLegendExtra={false}
         theme={chartTheme}
+        tooltip={{
+          headerFormatter: (d) => xAxisFormatter.convert(d.value),
```

Comment:
This does of course fail if there are non-matching types on the X axis, but I think we can't fix that case.